docs: clarify cloud and controller authentication story (#832)
juju/terraform-provider-juju: docs-rtd/howto/manage-controllers.md

(reference-an-externally-managed-controller)=
## Reference an externally managed controller

> To reference a controller that you've created outside of Terraform (because Terraform does not support controller bootstrap), in your `provider` definition add your controller address(es) and your controller authentication details. You can do this in one of 3 ways: using static credentials, using environment variables, or using the `juju` client. Note: The last method is supported only for regular Juju controllers. Across all the supported methods, for authentication with a regular Juju controller you must provide the username and password for a user, whereas for authentication with a JIMM controller you must provide the client ID and client secret for a service account (where the service account must be created through the external identity provider connected to the JIMM controller).

Reviewer: Feel free to adopt the changes/tweak/ignore.
```suggestion
To reference a controller that you've created outside of Terraform (because Terraform does not support controller bootstrap), in your `provider` definition add your controller address(es) and your controller authentication details. You can do this in one of 3 ways: using static credentials, using environment variables, or using the `juju` client. Note: The last method is not supported with JIMM. Across all the supported methods, when authenticating with Juju you must provide the username and password for a user, whereas for authentication with JIMM you must provide the client ID and client secret for a service account (where the service account must be created through the external identity provider connected to the JIMM controller).
```
I'd suggest avoiding the term "regular Juju controllers" as it could imply we have multiple non-regular Juju controllers. Instead I'd just say Juju or JAAS/JIMM.

Author: We started using the "regular" bit as a way to distinguish between Juju controllers added to JIMM _vs. not_ (aka 'regular'); see, e.g., https://documentation.ubuntu.com/terraform-provider-juju/latest/howto/manage-models/#for-a-regular-juju-controller. However, you're right, that is not what we actually mean here, so I'll drop "regular" here.
